clang/File.d

```diff
@@ -13,8 +13,13 @@ struct File
 {
     mixin CX;
 
-    @property string name ()
+    string name() const
     {
-        return toD(clang_getFileName(cx));
+        return toD(clang_getFileName(cast(void*) cx));
```

Comment:
Hard to say if it is absolutely necessary, but very convenient. And the user of the interface expects that getters that do not change the objects can be called on const object.

clang_getFileName do not modify its argument:



    CXString clang_getFileName(CXFile SFile) {
      if (!SFile)
        return cxstring::createNull();

      FileEntry *FEnt = static_cast<FileEntry *>(SFile);
      return cxstring::createRef(FEnt->getName());
    }

    class FileEntry {
      const char *Name;           // Name of the file.
     
      /* ... */

      const char *getName() const { return Name; }


    CXString createRef(StringRef String) {
      // If the string is not nul-terminated, we have to make a copy.

      // FIXME: This is doing a one past end read, and should be removed! For memory
      // we don't manage, the API string can become unterminated at any time outside
      // our control.

      if (!String.empty() && String.data()[String.size()] != 0)
        return createDup(String);

      CXString Result;
      Result.data = String.data();
      Result.private_flags = (unsigned) CXS_Unmanaged;
      return Result;
    }

(at least for now).